config.js

```diff
@@ -52,6 +51,7 @@ config.iota = {
     subservice: '/howto',
     providerUrl: 'http://localhost:4041',
     deviceRegistrationDuration: 'P1M',
+    defaultKey: '1234',
```

Comment:
conf.mqtt.defaultKey is referenced at https://github.com/telefonicaid/iotagent-json/blob/master/lib/iotaUtils.js#L48

Reply:
Right, maybe this explains the mystery of the missing test in CI. In any case I still don't understand why failure is on AMPQ only and not the others.

Maybe I need to keep it there in "mqtt" also, but note that if kept in both configs (mqtt and iota) we're back to the first problem 😕 . This means the workaround mentioned in #207 is not the way to go.

I may need some perspective here. If you have any suggestion please let me know.

Reply:
After fix this issue: https://github.com/telefonicaid/iotagent-json/issues/247 you will see a error trace about defaultKey is not fund. Could you please try it again?

Reply:
Sorry, I can't see how that fix and what you are asking connects to the problem being discussed. I'd be glad if you could help me understand.

Let's recap for a bit. The problem is, using a `config.js` as it is in the root of the project will not produce the expected results of the step-by-step (section "Provisioning a single device with the default API Key"). In particular, attribute values are not seen.

Maybe there's nothing wrong with the agent config, and what needs to be fixed is the the doc. I'm assuming the guide is right and the config is wrong, but please let me know if you think it's otherwise!

The config.js has defaultKey: '1234' only in the config.mqtt section.
People reported that the "workaround" was to move that definition to the config.iota section.
The workaround seems to be wrong for some tests in CI.

My question is, do you know which is the correct location of such a config?. 

Trying to answer your question, I tried the latest version (docker image tagged latest from 5 hours ago which I suppose brings the latest fix).
- Keeping the config.js as it is, it still "does not work".
- I removed all the definitions of defaultKey from the config.js being used to see if I got that message you mention with the fix but I didn't see anything (using loglevel DEBUG) :s.

Reply:
In order not blocking the PR, an option would be to leave the changes in config.js out and, once they get clarified (using https://github.com/telefonicaid/iotagent-json/issues/222 to hold dicussion), do it in a next PR.

What do you think?

Reply:
Sounds good to me :)

Reply:
Thus, NTC